From ba295c1f6672b891e14fe2929737329f01aa5fd2 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Fri, 25 Jul 2025 14:08:49 +0200
Subject: [PATCH] dnsmasq: Make log in ujail optional

Currently if you have ujail enabled and your logger hasn't started yet
when dnsmasq is being started, ujail fails to be created and dnsmasq
fails to start. So this patch makes the dependency on the logger
optional and allows dnsmasq with ujail to start even if there is no
logger daemon running.
---
 package/network/services/dnsmasq/files/dnsmasq.init | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/package/network/services/dnsmasq/files/dnsmasq.init b/package/network/services/dnsmasq/files/dnsmasq.init
index d2bcca46ed..3724e42ffe 100755
--- a/package/network/services/dnsmasq/files/dnsmasq.init
+++ b/package/network/services/dnsmasq/files/dnsmasq.init
@@ -1168,7 +1168,9 @@ dnsmasq_start()
 	[ -n "$instance_ifc" ] && network_get_device instance_netdev "$instance_ifc" &&
 		[ -n "$instance_netdev" ] && procd_set_param netdev $instance_netdev
 
-	procd_add_jail dnsmasq ubus log
+	local jail_features="ubus"
+	[ -e /dev/log ] && jail_features="$jail_features log"
+	procd_add_jail dnsmasq $jail_features
 	procd_add_jail_mount $CONFIGFILE $DHCPBOGUSHOSTNAMEFILE $DHCPSCRIPT $DHCPSCRIPT_DEPENDS
 	procd_add_jail_mount $EXTRA_MOUNT $RFC6761FILE $TRUSTANCHORSFILE
 	procd_add_jail_mount $dnsmasqconffile $dnsmasqconfdir $resolvdir $user_dhcpscript
-- 
2.50.1

